src/View/StringTemplate.php

```diff
@@ -323,4 +324,58 @@ protected function _formatAttribute($key, $value, $escape = true)
 
         return $key . '="' . ($escape ? h($value) : $value) . '"';
     }
+
+    /**
+     * Adds a class and returns a unique list either in array or space separated
+     *
+     * ### Options
+     *
+     * - `useIndex` if you are inputting an array with an 'element' other than 'class'.
+     *              Also setting to 'false' will manipulate the whole array (default is 'class')
+     *
+     * @param array|string $input The array or string to add the class to
+     * @param array|string $newClass the new class or classes to add
+     * @param string $useIndex if you are inputting an array with an 'element' other than 'class'.
+     *               Also setting to 'false' will manipulate the whole array (default is 'class')
+     * @return array|string
+     */
+    public function addClass($input, $newClass, $useIndex = 'class')
+    {
+        // NOOP
+        if (empty($newClass)) {
+            return $input;
+        }
+
+        if (is_string($useIndex) && is_array($input)) {
+            $class = Hash::get($input, $useIndex, []);
+        } else {
+            $class = $input;
+        }
+
+        // Convert and sanitise the inputs
+        if (!is_array($class)) {
+            if (is_string($class) && !empty($class)) {
+                $class = explode(' ', $class);
+            } else {
+                $class = [];
+            }
+        }
+
+        if (is_string($newClass)) {
+            $newClass = explode(' ', $newClass);
+        }
+
+        $class = array_unique(array_merge($class, $newClass));
+
+        if (is_string($useIndex)) {
```

Comment:
If we don't need it *now*, I prefer to not have 'extra' code. It adds complexity, requires more testing and more documentation.